chips/lpc55s6x/src/uart.rs

```diff
@@ -0,0 +1,990 @@
+    }
+
+    pub fn clear_status_flags_and_fifos(&self) {
+        self.registers.stat.write(
+            STAT::DELTACTS::SET
+                + STAT::FRAMERRINT::SET
+                + STAT::PARITYERRINT::SET
+                + STAT::RXBRK::SET,
+        );
+
+        self.registers
+            .fifocfg
+            .modify(FIFOCFG::EMPTYTX::SET + FIFOCFG::EMPTYRX::SET);
+    }
+}
+
+impl Configure for Uart<'_> {
+    fn configure(&self, params: Parameters) -> Result<(), ErrorCode> {
+        let clocks = self.clocks.get().ok_or(ErrorCode::OFF)?;
+        let flexcomm = self.flexcomm.get().ok_or(ErrorCode::OFF)?;
+        let clock_source = self.uart_clock_source.get();
+        let frg_id = FrgId::from_u32(self.instance.into()).ok_or(ErrorCode::INVAL)?;
+        clocks.setup_uart_clock(frg_id, clock_source);
+        flexcomm.configure_for_uart();
+
+        // --- Disable USART before configuration ---
+        self.registers.cfg.modify(CFG::ENABLE::CLEAR);
+
+        let clk = clocks.get_frg_clock_frequency(clock_source);
+        let brg_val = (clk / (16 * params.baud_rate)).saturating_sub(1);
+        if brg_val > 0xFFFF {
+            return Err(ErrorCode::INVAL); // Baud rate not possible
+        }
+        self.registers.brg.set(brg_val);
+
+        // --- Configure Frame Format (width, parity, stop bits) ---
+        let datalen = match params.width {
+            Width::Seven => CFG::DATALEN::_7BitDataLength,
+            Width::Eight => CFG::DATALEN::_8BitDataLength,
+            _ => return Err(ErrorCode::NOSUPPORT), // 6 and 9 bit not handled here
+        };
+
+        let paritysel = match params.parity {
+            Parity::None => CFG::PARITYSEL::NO_PARITY,
+            Parity::Odd => CFG::PARITYSEL::ODD_PARITY,
+            Parity::Even => CFG::PARITYSEL::EVEN_PARITY,
+        };
+
+        let stoplen = match params.stop_bits {
+            StopBits::One => CFG::STOPLEN::_1StopBit,
+            StopBits::Two => CFG::STOPLEN::_2StopBits,
+        };
+
+        // Write all configuration bits at once
+        self.registers.cfg.write(datalen + paritysel + stoplen);
+
+        // --- Configure and Enable FIFOs ---
+        // Clear any old data
+        self.registers
+            .fifocfg
+            .modify(FIFOCFG::EMPTYTX::SET + FIFOCFG::EMPTYRX::SET);
+        // Enable both TX and RX FIFOs
+        self.registers
+            .fifocfg
+            .modify(FIFOCFG::ENABLETX::SET + FIFOCFG::ENABLERX::SET);
+        // Set interrupt trigger levels.
+        self.registers
+            .fifotrig
+            .write(FIFOTRIG::TXLVL.val(1) + FIFOTRIG::RXLVL.val(0));
+
+        // --- Re-enable USART ---
+        self.registers.cfg.modify(CFG::ENABLE::SET);
+
+        // A short busy-wait loop is required to allow the peripheral clock
+        // to propagate and the internal logic to settle after being re-enabled
+        for _ in 0..1500 {
+            unsafe {
+                asm!("nop");
+            }
+        }
```

Comment:
Can you use https://github.com/tock/tock/blob/master/arch/cortex-m/src/support.rs#L12 instead?